lib/sidekiq_publisher/client.rb

```diff
@@ -7,7 +7,10 @@ class Client < Sidekiq::Client
     def bulk_push(items)
       payloads = items.map do |item|
         normed = normalize_item(item)
-        process_single(item["class"], normed) || nil
+        middleware.invoke(item["class"], normed, normed["queue"], @redis_pool) do
+          verify_json(normed) if respond_to?(:verify_json) # needed here as of v6.4.2, formerly done by normalize_item
```

Comment:
Out of curiosity, is there any harm (performance- or other-wise) in doing this twice if `normalize_item` used to do this, and now doesn't? That would mean in Sidekiq < 6.4.2, we'd do this twice, while in Sidekiq 6.4.2+, we'd run it once.

Reply:
The `verify_json` method doesn't exist until 6.4.2 -- it appears to contain the exact code that was pulled out of `normalize_item` in that version. So AFAICT there's no way to do this twice (except obviously literally calling the same method twice) because whatever version we're running against has only one way of invoking that code.

Reply:
Ah! So this actually will break on Sidekiq < 6.4.2, I think. Since our version constraint allows 5.0.4+, we might want to check for `process_single`, use it, or else do this more manual approach.

Reply:
The `respond_to?` check is intended to guard against breaking on lower versions (and all appraisals pass, such as [this one with sidekiq 5.0 and rails 5](https://app.circleci.com/pipelines/github/ezcater/sidekiq_publisher/115/workflows/f06e3810-c0a5-4895-81de-3f14b309bc5d/jobs/229/parallel-runs/0/steps/0-111) passes).

I thought about doing a proper gem version check like what's done in SidekiqPublisher::Compatibility. And (if I'm understanding you correctly) we could also apply either a gem version check or a `respond_to?` check to the fundamental fix, e.g.:

```
if respond_to?(:process_single) then process_single(...)
else middleware.invoke(...) { ... }
end || nil
```

...but all of those options added substantially more code to maintain and felt contrary to the idea of the limited, minimal fix.

Reply:
... well duh, I completely missed the `respond_to?` check! Disregard my obliviousness. 🤦🏻